https://issues.redhat.com/browse/ACM-10866 (#6581)
open-cluster-management/rhacm-docs: clusters/discovery/discovery_manage.adoc

[#managing-discovery]

> = Importing by using Discovery

Reviewer: Enabling a discovered cluster for management, filename **enable_discovered_cluster.adoc**

//since I mention ACM, should this be added to the new acm_integration folder instead? | MJ | 07/02

Automatically import {rosa} clusters into your {product-title} hub cluster with the `discovery-operator` for faster cluster management, without manually importing individual clusters.

*Required access:* Cluster administrator

=== Prerequisites
//does cluster admin cover this? Do we need ACM hub cluster if yes this needs to move to acm_integration
* You must have access to the namespaces that were used to configure the discovery service.

> * You must have permission to import your cluster to enable automatic import.

Reviewer: remove

Reviewer: Discovery is enabled by default. If you changed default settings, you need to enable Discovery.

[#managing-discovery]

> = Importing using the Discovery s

Author: ```suggestion
= Importing by using Discovery
```

[#managing-discovery]
= Managing Discovery services

> Review the following sections to learn more about detaching managed cluster instances, importing ROSA cluster by using a policy, and importing non-ROSA clusters.

Author: ```suggestion
Review the following sections to learn more about detaching managed cluster instances and importing non-ROSA clusters.
```

> The application management functions provide you with unified and simplified options for constructing and deploying applications and application updates. With these functions, your developers and DevOps personnel can create and manage applications across environments through channel and subscription-based automation. 

Author: ```suggestion
```

* <<detach-managed-cluster,Detaching managed clusters for Discovery>>
* <<import-non-rosa-clusters,Importing non-ROSA clusters automatically>>

> * <<import-rosa-policy,Importing ROSA cluster bu using a policy>>

Author: ```suggestion
```

[#detach-managed-cluster]
== Detaching managed clusters for Discovery

> Detach managaed clusters for Discovey to prevent reimporting automatically. Complete the following steps:

Author: ```suggestion
Detach managed clusters for Discovey to prevent reimporting automatically. Complete the following steps:
```

> Detach managaed clusters for Discovey to prevent reimporting automatically. Complete the following steps:

Author: ```suggestion
Detach managaed clusters for Discovey to prevent reimporting automatically. When you detach your cluster, it is also deleted. Complete the following steps:
```

//need to confirm steps
. Log in to your {product-title} cluster.
. Select the `DiscoverCluster` resource of the `ManagedCluster` instance that is managed by the hub cluster. 

> . To prevent reimport, add the following annotation, `discovery.open-cluster-management.io/previously-auto-imported: 'true'`.

Author: ```suggestion
. To detach your cluster, select the *Action menu* icon. 
. Select *Detach cluster*. The `discovery-operator` adds the following annotation, `discovery.open-cluster-management.io/previously-auto-imported: 'true'`.
```

+
Your `DiscoveredCluster` resource might resemble the following YAML:

+
[source,yaml]
----
apiVersion: discovery.open-cluster-management.io/v1
kind: DiscoveredCluster
metadata:
  annotations:
    discovery.open-cluster-management.io/previously-auto-imported: 'true'
----

> . To verify that the `DiscoveredCluster` resource is not reimported automatically, check for the following message during your reimport attempt:

Author: Make sure this is clear

+
[source,bash]
----
2024-06-12T14:11:43.366Z INFO reconcile	Skipped automatic import for DiscoveredCluster due to 
existing 'discovery.open-cluster-management.io/previously-auto-imported' annotation {"Name": "rosa-dc"}
----

. If you want to reimport the `DiscoveredCluster` resource automatically, you must remove the previously mentioned annotation.

[#import-non-rosa-clusters]
== Importing non-ROSA clusters automatically

> If you want to automatically import a non-ROSA cluster, the import does not initiate. You must import other `DiscoveredCluster` resources manually. You must directly apply Kubernetes configurations to the other types of `DiscoveredCluster` resources. If you enable the `importAsManagedCluster` field from the `DiscoveredCluster` resource, it is prevented by the Discovery webhook. 

Author: ```suggestion
If you want to automatically import a non-ROSA cluster, the import does not initiate. You must import other infrastructure provider `DiscoveredCluster` resources manually. You must directly apply Kubernetes configurations to the other types of `DiscoveredCluster` resources. If you enable the `importAsManagedCluster` field from the `DiscoveredCluster` resource, it is not imported due to the Discovery webhook. 
```

> *Note:* If there is a `MultiClusterEngineHCP` type for the `DiscoveredCluster` resource you can automatically import it. However, the `DiscoveredCluster` resource is provided through only HyperShift. For more details see, 

Author: Might remove, but keeping for now.

Author: HyperShift also has a policy that can be configured to automatically import clusters. Might want to point to that info if it exists.

metadata:
  annotations:
    discovery.open-cluster-management.io/previously-auto-imported: 'true'
----

. To verify that the `DiscoveredCluster` resource is not reimported automatically, check for the following message in the `DiscoveredCluster` logs:

+
[source,bash]
----
2024-06-12T14:11:43.366Z INFO reconcile	Skipped automatic import for DiscoveredCluster due to 
existing 'discovery.open-cluster-management.io/previously-auto-imported' annotation {"Name": "rosa-dc"}
----

. If you want to reimport the `DiscoveredCluster` resource automatically, you must remove the previously mentioned annotation.

[#import-non-rosa-clusters]
== Importing non-{rosa} clusters automatically

If you want to automatically import a non-ROSA cluster, the import does not initiate. You must import other infrastructure provider `DiscoveredCluster` resources manually. You must directly apply Kubernetes configurations to the other types of `DiscoveredCluster` resources. If you enable the `importAsManagedCluster` field from the `DiscoveredCluster` resource, it is not imported due to the Discovery webhook. See, link:../discovery/config_ui.adoc#import-discovered[Import discovered clusters from the console].

> //*Note:* If there is a `MultiClusterEngineHCP` type for the `DiscoveredCluster` resource you can automatically import it. However, the `DiscoveredCluster` resource is provided through only HyperShift. For more details see, 

Author: Found the file that mentions the automatic import for Hosted control plane clusters: https://github.com/stolostron/rhacm-docs/blob/2.11_stage/clusters/hosted_control_planes/create_hosted_aws.adoc

[#managing-discovery]
= Importing by using Discovery
//since I mention ACM, should this be added to the new acm_integration folder instead? | MJ | 07/02

Automatically import {rosa} clusters into your {product-title} hub cluster with the `discovery-operator` for faster cluster management, without manually importing individual clusters.

*Required access:* Cluster administrator

=== Prerequisites
//does cluster admin cover this? Do we need ACM hub cluster if yes this needs to move to acm_integration
* You must have access to the namespaces that were used to configure the discovery service.
* You must have permission to import your cluster to enable automatic import.

Review the following sections to learn more about importing and managing discovered clusters:

* <<import-discovered-auto-rosa,Importing discovered clusters automatically for {rosa} clusters>>
* <<detach-managed-cluster,Detaching managed clusters for Discovery>>
* <<import-non-rosa-clusters,Importing non-{rosa} clusters automatically>>

[#import-discovered-auto-rosa]
== Importing discovered {rosa} clusters automatically  

> //need content here

Author: ```suggestion
The following procedure is an example of how to import your discovered {rosa} clusters automatically by using the `discovery-operator`.
```
